plugins/plugins.go

```diff
@@ -982,3 +1030,46 @@ func (m *Manager) RegisterNDCacheTrigger(trigger func(bool)) {
 	defer m.mtx.Unlock()
 	m.registeredNDCacheTriggers = append(m.registeredNDCacheTriggers, trigger)
 }
+
+func (m *Manager) sendOPAUpdateLoop(ctx context.Context, uploadDuration time.Duration) {
+	ticker := time.NewTicker(uploadDuration)
+	mr.New(mr.NewSource(time.Now().UnixNano()))
+
+	ctx, cancel := context.WithCancel(ctx)
+
+	var opaReportNotify bool
+
+	for {
+		select {
+		case <-m.opaReportNotifyCh:
+			opaReportNotify = true
+		case <-ticker.C:
+			ticker.Stop()
+
+			if opaReportNotify {
+				m.reporter.RegisterGatherer("min_compatible_version", func(_ context.Context) (any, error) {
```

Comment:
The registration should happen outside if this loop. In that loop, we'll invoke all registered gatherer functions and aggregate their results.

Reply:
I've moved registration outside the loop and after we create the reporter instance. 

> In that loop, we'll invoke all registered gatherer functions and aggregate their results.

In this case we only have one gatherer function and we're not adding any public methods on the manager to add gatherers so not sure what you mean here. In the loop we call send on the reporter which will invoke the gatherer functions.

Reply:
> In this case we only have one gatherer function and we're not adding any public methods on the manager to add gatherers so not sure what you mean here.

Yeah, that's correct right now. I think ideally, we'd open this up, so wrapping code (😉 EOPA) can register extra telemetry bits.

Reply:
Updated to include a method on the manager to allow callers to register extra telemetry gatherers.